### PR TITLE
Update and add Functions test coverage

### PR DESCRIPTION
**Pull request by unfulvio-godaddy**
# Summary 

This PR:

* Refactors and adds more unit test coverage for the `Functions` class
* Adds integration test coverage for `userFunction`, `echoFunction` and `passthruFunction` methods
* Test coverage bumped to ~60%

Had to switch PHP 7.3 tests off due to [this issue](https://github.com/10up/wp_mock/issues/212)

## Contributor checklist 

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly 
- [x] I have added tests to cover changes introduced by this pull request
- [x] All new and existing tests pass

### Reviewer checklist 

 

- [ ] Code changes review
- [ ] Documentation changes review
- [ ] Unit tests pass
- [ ] Static analysis passes

### CODE REVIEW

**Comment by unfulvio-godaddy** [comment]
looks like there's an issue with Coveralls and/or PhpUnit not generating coverage or not running tests - I had to tweak the phpunit conf file to make sure integration tests were run but maybe this triggers an issue when run in CI

File: tests/Mocks/SampleSubClass.php [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 class SampleSubClass extends SampleClass
 {
-    public function action() : void
+    public function action(): void
```
Comment (nmolham-godaddy) [inline tests/Mocks/SampleSubClass.php:7]: Do changes like this one intentional? or my accident via the IDE?

Comment (unfulvio-godaddy) [reply tests/Mocks/SampleSubClass.php:7]: I don't think you have accidentally committed that - seems we are using that return type declaration format consistently via PHPCS

**Review by nmolham-godaddy — APPROVED** [review]
LGTM

**Pull request merged**